### PR TITLE
Don't change `RID` when changing `viewport_path` in `ViewportTexture`

### LINKED ISSUE #77207: Changing the `viewport_path` of the `ViewportTexture` in a material renders it white
### Godot version

v4.1.dev.custom_build [18112ba36]

### System information

Linux Mint 21.1, NVIDIA GeForce GTX 1050 Ti,  Vulkan API 1.3.205

### Issue description

This is a regression from #64388. 

Textures seem to be used differently in 2D and 3D. They are rendered using the `RID` in 3D, and changing the `viewport_path` will cause the `RID` to change.

https://github.com/godotengine/godot/assets/30386067/ad393fe8-50c1-4a89-870d-58d4e224bcc6

### Steps to reproduce

1. Download and import MRP.
2. Open 1.tscn, and try to switch the `viewport_path` of the `MeshInstance3D:mesh:material:albedo_texture` in the MeshInstance3D node.
3. After switching, the MeshInstance3D is rendered as white in the editor;
4. Close and then reopen 1.tscn.

### Minimal reproduction project

[ViewportTexture-change-viewport_path.zip](https://github.com/godotengine/godot/files/11509010/ViewportTexture-change-viewport_path.zip)

### PR DESCRIPTION
**Pull request by Rindbee**
When changing `viewport_path`, the `proxy`'s target is changed to a new placeholder.

Add a flag `vp_changed` to prevent calling `setup_local_to_scene()` (mainly called while toggling `resource_local_to_scene`) when the target viewport has not changed.

Fix #77207.

### CODE REVIEW

**Review by KoBeWi — APPROVED** [review]
I confirmed that it fixes the issue, but I can't fully understand the changes (someone from core team would be more qualified). It doesn't seem to make things worse.

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -89,20 +93,17 @@ void ViewportTexture::set_viewport_path_in_scene(const NodePath &p_path) {
...
-		RS::get_singleton()->free(proxy_ph);
-	}
-	if (proxy.is_valid()) {
-		RS::get_singleton()->free(proxy);
+	if (proxy.is_valid() && proxy_ph.is_null()) {
+		proxy_ph = RS::get_singleton()->texture_2d_placeholder_create();
+		RS::get_singleton()->texture_proxy_update(proxy, proxy_ph);
 	}
-	proxy_ph = RID();
-	proxy = RID();
```
Comment (RandomShaper) [inline scene/main/viewport.cpp:105]: Wouldn't it be enough to just remove this?

Aside, it seems that the original intent is that `proxy` and `proxy_ph` are both either set or null and the changes in this PR seem to challenge that.

Comment (Rindbee) [reply scene/main/viewport.cpp:105]: What is considered in `_setup_local_to_scene()` is actually the case of setting the target viewport for the first time. The case of changing the target viewport is actually not considered.

The main consideration here is when the `viewport_path` is reset in the editor. 

`proxy` may be valid when reset. Additionally, `viewport_path` has the potential to be reset to an empty path, which means that `_set_viewport_path_in_scene()` will not be called, and it should disconnect from the previous target viewport, this needs to be done in `set_viewport_path_in_scene()`.

In other words, `set_viewport_path_in_scene()` should reset the state, making the `ViewportTexture` look like it was never set.

Comment (RandomShaper) [reply scene/main/viewport.cpp:105]: I think I get it now. Thanks for the explanation.

Last thing I would ask is maybe one/some comment(s) in the code, maybe describing what `vp_changed` is for.

File: scene/main/viewport.cpp [hunk 2]
```diff
@@ -61,6 +61,11 @@
 #include "servers/rendering/rendering_server_globals.h"
 
 void ViewportTexture::setup_local_to_scene() {
+	// For the same target viewport, setuping is only allowed once to prevent multiple free or multiple creations.
```
Comment (akien-mga) [inline scene/main/viewport.cpp:64]: I'd go with just "setup", or "setting up", to avoid inventing a word ;)

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
We can consider this for 4.0.x again after the 4.1 release, as that would serve as additional testing for this and https://github.com/godotengine/godot/pull/64388. At that point I would ask to create a dedicated 4.0 PR based on the master version of the fixes to avoid cherry-picking conflicts.

So for that reason I'm removing the cherry-pick label.
